### ISSUE TITLE
Out/Ref parameters have to be local.

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#1202]
[Bridge Example](http://live.bridge.net/#2835645f5d174e71f08d42209206b631)
[.Net Example](https://dotnetfiddle.net/YqKrNU)

In Bridge, it shows the error
```
C# to JavaScript Compilation Failed

Bridge.Contract.EmitterException: Only local variables can be passed by reference
   at Bridge.Translator.AbstractEmitterBlock.AddLocals(IEnumerable`1 declarations, AstNode statement)
   at Bridge.Translator.VisitorMethodBlock.VisitMethodDeclaration(MethodDeclaration methodDeclaration)
   at Bridge.Translator.VisitorMethodBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.VisitMethodDeclaration(MethodDeclaration methodDeclaration)
   at Bridge.Translator.MethodBlock.EmitMethodsGroup(List`1 group)
   at Bridge.Translator.MethodBlock.EmitMethods(Dictionary`2 methods, Dictionary`2 properties, Dictionary`2 operators)
   at Bridge.Translator.MethodBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.ClassBlock.EmitStaticBlock()
   at Bridge.Translator.ClassBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.EmitBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.Emit()
   at Bridge.Translator.Translator.Translate()
   at Live.LiveTranslator.Translate(String& output)
   at Live.TranslateHandler.Translate(String csCode)
```
Code: 
```csharp
/// <summary>
/// Creates a simple class
/// </summary>
public class App
{
    public static void Cool (out int value)
    {
        value = 3;
    }
    static int v2;
    // Use the [Ready] attribute to automaically
    // run this method when the page is ready.
    [Ready]
    public static void Main()
    {
        Cool(out v2);
        Console.WriteLine(v2);
    }
}
```
The only difference between the 2 fiddles is the removed `[Ready]`.

**Comment by vladsch** [comment]
Yes, at this moment, Bridge supports out/ref modifiers for local variables only
We will try to remove that limitation in next releases

**Comment by michaelcheers** [comment]
@vladsch,
This hasn't been fixed, Live Bridge says that it is version 1.13 (the milestone for this pull request). [Live Bridge](http://live.bridge.net/#2835645f5d174e71f08d42209206b631).

**Comment by LeonidVeriga** [comment]
@michaelcheers, thank you for reporting [this](https://github.com/bridgedotnet/Bridge/issues/1202#issuecomment-219291559).The fix in 1.13 actually works. It is a Live Bridge problem, we will fix soon.
